examples/stats_models/regression.jl

```diff
@@ -0,0 +1,243 @@
+
+using SemanticModels
+using Random
+using LightGraphs
+using MetaGraphs
+using DataFrames
+using GraphPlot
+using DifferentialEquations
+using Plots
+using MultivariateStats
+using Statistics
+
+mutable struct DepVar{T}
+    value::T
+end
+
+mutable struct IndepVar{T}
+    value::T
+end
+
+# to allow transformations that feature numeric elements/vectors 
+mutable struct Other{T}
+    value::T
+end
+
+
+const Var = Union{DepVar, IndepVar}
+const RegComponents = Union{DepVar, IndepVar, Other}
+
+
+mutable struct OpElementTypes{a, b}
+    vec_a::a
+    vec_b::b
+end
+
+function var_type_check(a::Var, b::Other)
+    return isa(a, DepVar) ? DepVar : IndepVar
```

Comment:
This type checks could instead be done with multiple dispatch:
```julia
var_type_check(a::DepVar, b::Other) = DepVar
var_type_check(a::IndepVar, b::Other) = IndepVar
...
```
This is more idiomatic than branching on `isa()`.

Reply:
fixed.